### PR TITLE
Increased stability of metric asynchronous execution

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/telemetry/metric.go [hunk 1]
```diff
@@ -163,37 +171,50 @@ func (m *MetricController) Run(ctx context.Context, mapOfTcpInfo *ebpf.Map) {
...
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		default:
+			rec := ringbuf.Record{}
 			if err := reader.ReadInto(&rec); err != nil {
 				log.Errorf("ringbuf reader FAILED to read, err: %v", err)
-				continue
+				return
```
Comment (YaoZengzeng) [inline pkg/controller/telemetry/metric.go:182]: why did it change from `continue` to `return`? 

`getConnectDataFromRingBuf` run as goroutine and don't handle errors. Event if it fails, only the goroutine exits and the rest of the system is completely unaware. 

We need a fault-tolerant mechanism or an overall exit after failure

File: pkg/controller/telemetry/metric.go [hunk 2]
```diff
@@ -163,37 +171,50 @@ func (m *MetricController) Run(ctx context.Context, mapOfTcpInfo *ebpf.Map) {
...
 			default:
-				log.Errorf("get connection info failed: %v", err)
-				continue
+				log.Errorf("Unsupported IP family")
+				return
+			}
+
+			if err != nil {
+				log.Debugf("get data failed: %v", err)
+				return
```
Comment (YaoZengzeng) [inline pkg/controller/telemetry/metric.go:217]: ditto

File: pkg/controller/telemetry/metric.go [hunk 3]
```diff
@@ -297,7 +318,7 @@ func (m *MetricController) getWorkloadByAddress(address []byte) (*workloadapi.Wo
 	networkAddr.Address, _ = netip.AddrFromSlice(address)
 	workload := m.workloadCache.GetWorkloadByAddr(networkAddr)
 	if workload == nil {
-		log.Warnf("get workload from ip %v FAILED", address)
+		log.Debugf("get workload from ip %v FAILED", address)
```
Comment (YaoZengzeng) [inline pkg/controller/telemetry/metric.go:321]: Why did it change from `Warnf` to `Debugf`? Does this happen frequently? And is it normal that workload can't be found from IP?

**Review by hzxuzhonghu — COMMENTED** [review]
We still need this async processing, there are crirical section protected during metric building

File: pkg/controller/telemetry/metric.go [hunk 4]
```diff
@@ -350,14 +371,12 @@ func buildServiceMetric(dstWorkload, srcWorkload *workloadapi.Workload, dstPort
 			}
 		}
 		if namespacedhost == "" {
-			log.Infof("can't find service correspond workload: %s", dstWorkload.Name)
+			log.Debugf("can't find service correspond workload: %s", dstWorkload.Name)
```
[the remark this answers is not in the document]
Comment (LiZhenCheng9527) [reply pkg/controller/telemetry/metric.go:374]: This error message has been removed from the main branch.

File: pkg/controller/telemetry/metric.go (hunk 3 above)
Comment (LiZhenCheng9527) [reply pkg/controller/telemetry/metric.go:321]: i think this is not error

**Comment by LiZhenCheng9527** [comment]
That's no longer necessary.

**Pull request closed without merging**